tests/dist.rs

```diff
@@ -32,11 +32,10 @@ use crate::mock::{MockComponentBuilder, MockFile, MockInstallerBuilder};
 
 const SHA256_HASH_LEN: usize = 64;
 
+pub type EditChannel = dyn Fn(/* date */ &str, &mut MockChannel);
+
 // Creates a mock dist server populated with some test data
-pub fn create_mock_dist_server(
-    path: &Path,
-    edit: Option<&dyn Fn(&str, &mut MockChannel)>,
-) -> MockDistServer {
+pub fn create_mock_dist_server(path: &Path, edit: Option<&EditChannel>) -> MockDistServer {
```

Comment:
So EditChannel is just defining the function signature that these two functions use as a closure.

We're separating out two things that are tightly coupled. I don't think that increases readability: it increases cognitive load.

Please allow() it at the module level. (I have a PR in flight that refactors this test code towards immutability, so the whole edit concept is going to fade anyhow).